docs/dev/plotting.md

```diff
@@ -0,0 +1,108 @@
+# Recommended interface and behavior of a `plot` method
+
+Here we recommend the input arguments, return value, and behavior of the
+`plot` method of a class.
+
+## Requirements
+
+1. **Convenience to interactive users.** This is the highest priority.
+    Compared to being called in a batch script as a library (for composing
+    more complicated plots or other purposes), the `plot` method is mainly
+    used in interactive sessions like ipython, jupyther, colab, PyCharm,
+    and python interpreter.
+1. **Plot is customizable.** The plot should be customizable by the user after
+    `plot` returns. This is important because user may need to change the look
+    for presentation, paper, or just the style they prefer. One plot style does
+    not fit all.
+1. **No unnecessary messages in interactive sessions.** It should not produce
+    any warning/error messages in normal usages in an interactive sessions.
+    See [#1890](https://github.com/quantumlib/Cirq/issues/1890#issue-473510953)
+    for an example of such message.
+1. **No popups during tests.** It should not produce any pop-up windows during
+    tests.
+
+## Recommendation
+
+The `plot` method must produce a plot when there is no arguments in an
+interactive session. The recommended way to achieve that is illustrated in the
+example below.
+
+```python
+class Foo:
+    ...
+    def plot(self, ax: plt.Axes=None, **plot_kwargs: Any) -> plt.Axes:
```

Comment:
Add an `import matplotlib.pyplot as plt` line. The example testing test is failing because `plt` is not imported at this point.

Reply:
Done, though the tests are green before the change.

Reply:
Good catch. That was a bug in the tests. Not it did fail before the move to a subdir of docs. Fixed in https://github.com/quantumlib/Cirq/pull/2307